src/components/views/projections/CustomLinkWidget.js

```diff
@@ -1,63 +1,265 @@
+    const {point, angle, pointId, linkId, pointlength} = props;
 
-export class CustomLinkWidget extends React.Component {
+    return (
+        <g className="arrow" transform={'translate(' + (point.x) + ', ' + (point.y) + ')'}>
+            <g style={{transform: 'rotate(' + angle + 'deg)'}}>
+                <polyline
+                    points={`0,0 ${pointlength},${pointlength} 0,${pointlength*2}`}
+                    {
+                        ...projectionLinkArrow
+                    }
+                    data-id={pointId}
+                    data-linkid={linkId}
+                />
+            </g>
+        </g>
+    );
+};
+
+
+class CustomLink extends React.Component {
     constructor(props) {
         super(props);
-        this.percent = 0;
+
+        this.path = React.createRef();
     }
 
     componentDidMount() {
         this.mounted = true;
-        this.callback = () => {
-            if (!this.circle || !this.path) {
-                return;
-            }
+    }
 
-            this.percent += 2;
-            if (this.percent > 100) {
-                this.percent = 0;
-            }
+    componentWillUnmount() {
+        this.mounted = false;
+    }
 
-            let point = this.path.getPointAtLength(this.path.getTotalLength() * (this.percent / 100.0));
+    shouldComponentUpdate() {
+        return true;
+    }
 
-            this.circle.setAttribute('cx', '' + point.x);
-            this.circle.setAttribute('cy', '' + point.y);
+    render() {
+        return (
+            <g>
+                <path
+                    ref={this.path}
+                    {
+                        ...projectionLink
+                    }
+                    d={this.props.path}
+				/>
+            </g>
+        );
+    }
+}
 
-            if (this.mounted) {
-                requestAnimationFrame(this.callback);
-            }
-        };
-        requestAnimationFrame(this.callback);
+
+
+export class CustomLinkWidget extends DefaultLinkWidget {
+    generateArrow(point, angle, pointId, linkId, pointlength) {
+        return (
+            <CustomLinkArrowWidget
+                key={`link-arrow-${pointId}`}
+                point={point}
+                angle={angle}
+                linkId={linkId}
+                pointId={pointId}
+                pointlength={pointlength}
+                colorSelected={this.props.link.getOptions().selectedColor}
+                color={this.props.link.getOptions().color}
+            />
+        );
     }
 
-    componentWillUnmount() {
-        this.mounted = false;
+    generateLinePath(firstPoint, lastPoint) {
+        if (firstPoint.x <= lastPoint.x) {
+            return `M${firstPoint.x + 15},${firstPoint.y} L ${lastPoint.x},${lastPoint.y}`;
+        } else {
+            return `M${firstPoint.x - 15},${firstPoint.y} L ${lastPoint.x},${lastPoint.y}`;
+        }
     }
 
     render() {
+        //ensure id is present for all points on the path
```

Comment:
@vidhya-metacell can you explain what you are doing here and below? seems a little complex to me

Reply:
@zsinnema , we are using our custom svg's for both projection link line and projection link arrow head to match with figma. So for calculating the position points to display these two @ddelpiano did all these calculations for points as well as the angle of the arrow head, in order to maintain the position of these links while we move the nodes on the canvas. May be @ddelpiano can explain better, if there any further questions on this.

Reply:
@ddelpiano I suggest to add comments to the calculations and move it to somewhere "generic"? Is this something generic special for PsyNeuLinkView or is it generic for MetaDiagram?

Reply:
@ddelpiano and also could you please remove the "fixed" constants like `pointlength` etc and replace it by either something dynamic calculated (depending on the source and target objects) or a configuration item